svalbard/xmlbeans/src/main/java/org/n52/svalbard/encode/InsertObservationRequestEncoder.java

```diff
@@ -70,6 +74,26 @@ private void addObservations(List<OmObservation> observations, InsertObservation
         }
     }
 
+    private void addExtensions(Extensions extensions, InsertObservationType insertObservation)
+            throws EncodingException {
+        if (extensions == null || extensions.isEmpty()) {
+            return;
+        }
+        final List<EncodingException> thrownExceptions = new LinkedList<>();
+        extensions.getExtensions().stream().forEach(o -> {
+            try {
+                if (thrownExceptions.isEmpty()) {
+                    insertObservation.addNewExtension().set(encodeObjectToXml(SwesConstants.NS_SWES_20, o));
+                }
+            } catch (EncodingException e) {
+                thrownExceptions.add(e);
+            }
+        });
+        if (!thrownExceptions.isEmpty()) {
+            throw thrownExceptions.get(0);
+        }
+    }
+
```

Comment:
Streams are a nice feature, but are utterly broken in combination with exceptions... This is much simpler:
```java
private void addExtensions(Extensions extensions, InsertObservationType insertObservation)
            throws EncodingException {
  if (extensions == null || extensions.isEmpty()) {
    return;
  }
  for (Extension<?> o : extensions.getExtension()) {
    insertObservation.addNewExtension().set(encodeObjectToXml(SwesConstants.NS_SWES_20, o));
  }
}